packages/core/src/browser/view-container.ts

```diff
@@ -623,25 +625,65 @@ export class ViewContainerPart extends BaseWidget {
         };
     }
 
+    protected onResize(msg: Widget.ResizeMessage): void {
+        if (this.wrapped.isAttached && !this.collapsed) {
+            MessageLoop.sendMessage(this.wrapped, Widget.ResizeMessage.UnknownSize);
+        }
+        super.onResize(msg);
+    }
+
+    protected onUpdateRequest(msg: Message): void {
+        if (this.wrapped.isAttached && !this.collapsed) {
+            MessageLoop.sendMessage(this.wrapped, msg);
+        }
+        super.onUpdateRequest(msg);
+    }
+
     protected onAfterAttach(msg: Message): void {
         if (!this.wrapped.isAttached) {
-            Widget.attach(this.wrapped, this.body);
+            MessageLoop.sendMessage(this.wrapped, Widget.Msg.BeforeAttach);
+            // tslint:disable-next-line:no-null-keyword
+            this.body.insertBefore(this.wrapped.node, null);
```

Comment:
what does this do? ;-)

Reply:
do you mean `null` is not necessary? otherwise it appends a widget to the part body

Reply:
thanks!